lib/ruby_lsp/ruby_lsp_rails/server.rb

```diff
@@ -19,6 +19,7 @@ def send_message(message)
       # Log a message to the editor's output panel
       def log_message(message)
         $stderr.puts(message)
+        send_message({ result: nil })
```

Comment:
This is indeed the right solution as we need to return something even when errors occur. But this is not the right place to put it.

The goal of the `log_message` method is to allow both the server and its add-ons to send debug messages to the output tab, even if disconnected from providing a result.

If a database error was to occur in a notification, where no response is expected, then we'd be writing a nil result to the pipe, which wouldn't be read by the notification. The next request would then incorrectly read it and we'd end up with the request/response cycle out of sync.

We need to move this line to where error handling happens, so that `log_message` is not coupled with returning responses to the editor. And this result can only be returned when rescuing errors for requests - not for notifications.

Reply:
Ah yes, I wasn't considering usage from add-ons.

https://github.com/Shopify/ruby-lsp-rails/pull/517